### ISSUE TITLE
SpoCon Crashes on Play

### ISSUE THREAD

**Issue by txoof** [spocon/spocon#55]
**Describe the bug**
SpoCon crashes when a playback is transferred to the SpoCon device

**Platform**
PRETTY_NAME="Raspbian GNU/Linux 11 (bullseye)"
VERSION="11 (bullseye)"
ID=raspbian
Linux 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux
SpoCon Version: librespot-java-api-v1.6.2.jar

**To Reproduce**
1. Fresh install of Bullseye32 Lite from RaspberryPi Imager
2. Install SpoCon using `curl -sL https://spocon.github.io/spocon/install.sh | sh`
3. Ensure spocon is running:
```$ sudo systemctl status spocon
● spocon.service - SpoCon
     Loaded: loaded (/lib/systemd/system/spocon.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2023-05-13 16:33:26 CEST; 3min 10s ago
    Process: 1837 ExecStartPre=/bin/sh -c until ping -c1 spotify.com; do sleep 5; done; (code=exited, st>
   Main PID: 1839 (java)
      Tasks: 29 (limit: 1599)
        CPU: 9.926s
     CGroup: /system.slice/spocon.service
             └─1839 /usr/bin/java -jar /opt/spocon/librespot-java-api-v1.6.2.jar
``` 
4. Using phone or desktop client, transfer playback to SpoCon device
5. Spocon crashes with log output below

**Expected behavior**
SpoCon should take the transferred playback, begin playing and not crash.

**Logs**
```
May 13 17:00:20 livingroom java[3785]: 2023-05-13 17:00:20,735 TRACE ZeroconfServer:536 - Handling request: GET /?action=getInfo&version=2.7.1 HTTP/1.1, headers: {Keep-Alive=0, Connection=keep-alive, User-Agent=Spotify/120900743 OSX_ARM64/0 (Macmini9,1), Host=192.168.1.192, Accept-Encoding=gzip, Content-Type=application/x-www-form-urlencoded}
May 13 17:00:26 livingroom java[3785]: 2023-05-13 17:00:26,904 TRACE ZeroconfServer:536 - Handling request: GET /?action=getInfo&version=2.7.1 HTTP/1.1, headers: {Keep-Alive=0, Connection=keep-alive, User-Agent=Spotify/120900743 OSX_ARM64/0 (Macmini9,1), Host=192.168.1.192, Accept-Encoding=gzip, Content-Type=application/x-www-form-urlencoded}
May 13 17:00:26 livingroom java[3785]: 2023-05-13 17:00:26,921 TRACE ZeroconfServer:536 - Handling request: POST / HTTP/1.1, headers: {Keep-Alive=0, Connection=keep-alive, User-Agent=Spotify/120900743 OSX_ARM64/0 (Macmini9,1), Host=192.168.1.192, Accept-Encoding=gzip, Content-Length=779, Content-Type=application/x-www-form-urlencoded}
May 13 17:00:27 livingroom java[3785]: 2023-05-13 17:00:27,179 INFO  ZeroconfServer:345 - Accepted new user from Aaron’s Mac mini. {deviceId: 497c197e11e456fa6ecca32333ed4fc720925176}
May 13 17:00:27 livingroom java[3785]: 2023-05-13 17:00:27,418 INFO  ApResolver:90 - Loaded aps into pool: {accesspoint=[ap-gew4.spotify.com:4070, ap-gew4.spotify.com:443, ap-gew4.spotify.com:80, ap-guc3.spotify.com:4070, ap-gew1.spotify.com:443, ap-gue1.spotify.com:80], dealer=[gew4-dealer.spotify.com:443, guc3-dealer.spotify.com:443, gew1-dealer.spotify.com:443, gue1-dealer.spotify.com:443], spclient=[gew4-spclient.spotify.com:443, guc3-spclient.spotify.com:443, gew1-spclient.spotify.com:443, gue1-spclient.spotify.com:443]}
May 13 17:00:27 livingroom java[3785]: 2023-05-13 17:00:27,458 DEBUG TimeProvider:90 - Loaded time offset from NTP: -3ms
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,069 INFO  Session:137 - Created new session! {deviceId: 497c197e11e456fa6ecca32333ed4fc720925176, ap: ap-gew4.spotify.com:443, proxy: false}
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,573 INFO  Session:331 - Connected successfully!
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,677 TRACE Session:1333 - Session.Receiver started
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,711 INFO  CacheManager:95 - There are 0 cached entries.
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,724 TRACE AsyncProcessor:52 - AsyncProcessor{dealer-worker} has started
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,731 TRACE AsyncProcessor:52 - AsyncProcessor{event-service-sender} has started
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,734 INFO  Session:1415 - Skipping SecretBlock
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,736 INFO  Session:1387 - Received LicenseVersion: 0
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,738 INFO  Session:1377 - Received CountryCode: NL
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,742 DEBUG TokenProvider:67 - Token expired or not suitable, requesting again. {scopes: [playlist-read], oldToken: null}
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,769 TRACE MercuryClient:156 - Send Mercury request, seq: 1, uri: hm://keymaster/token/authenticated?scope=playlist-read&client_id=65b708073fc0480ea92a077233ca87bd&device_id=497c197e11e456fa6ecca32333ed4fc720925176, method: GET
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,774 TRACE Session:776 - Parsed product info: {remote-control=6, publish-activity=0, audio-preview-url-template=https://p.scdn.co/mp3-preview/{id}, incognito_mode_timeout=21600, ad-formats-preroll-video=0, type=premium, ab_recently_played_feature_time_filter_threshold=com.spotify.gaia=30,driving-mode=120,spotify%3Ainternal%3Astartpage=30, license-agreements=, publish-playlist=0, buffering-strategy=0, expiry=1, ab-desktop-hide-follow=0, ab-collection-hide-unavailable-albums=0, capper-profile=, payments-initial-campaign=default, profile-image-upload=1, enable-annotations=2, ab-ad-player-targeting=1, india-experience=0, video-wifi-initial-bitrate=800000, ab-mobile-running-onlymanualmode=only-manual, playlist-annotations-markup=0, prefetch-keys=1, allow-override-internal-prefs=0, prefetch-window-max=2, send-email=0, ad-formats-video-takeover=1, ab-collection-bookmark-model=1, pause-after=0, local-files-import=0, video-manifest-url=https://spclient.wg.spotify.com/manifests/v6/{type}/sources/{source_id}/options/supports_drm, ab-sugarpills-sanity-check=0, playlist-folders=1, arsenal_country=1, track-cap=0, ab-mobile-running-tempo-detection=Control, is_email_verified=1, shows-collection=1, enable-annotations-read=0, addon-hifi=0, mobile-browse=0, payments-locked-state=0, employee-free-opt-in=0, shows-collection-jam=0, ab-browse-music-tuesday=1, offline=1, streaming=1, fb-info-confirmation=control, audio-quality=1, wanted-licenses=, lastfm-session=, ab-desktop-playlist-annotation-edit=1, ab-mobile-discover=0, widevine-license-url=https://spclient.wg.spotify.com/widevine-license/v1/video/license, key-caching-max-count=10000, hifi-eligible=0, ad-session-persistence=1, video-initial-bitrate=200000, ab-play-history=0, created_by_partner=, mobile-payment=0, payment-state=, key-caching-max-offline-seconds=1800, ugc-abuse-report-url=https://support.spotify.com/abuse/?uri={uri}, shuffle-algorithm=1, use-pl3=0, image-url=https://i.scdn.co/image/{file_id}, use-playlist-app=0, enable-autostart=1, metadata-link-lookup-modes=0, multiuserplan-current-size=2, shuffle=0, instant-search=0, instant-search-expand-sidebar=0, use-playlist-uris=0, user-profile-show-invitation-codes=0, multiuserplan-member-type=FAMILY_MANAGER, ab-ad-requester=1, nft-disabled=1, ab-collection-union=1, key-memory-cache-mode=1:15,300, libspotify=1, high-bitrate=1, head-file-caching=1, radio=1, key-caching-auto-offline=0, explicit-content=1, video-cdn-sampling=1, ab-android-push-notifications=1, unrestricted=1, use-fb-publish-backend=2, app-developer=0, buffering-strategy-parameters=0.8:0.2:0.0:0.0:0.0:0.0:1.0:10:10:2000:10000:10485760, enable-gapless=1, ab-playlist-extender=5, enable-crossfade=1, ad-persist-reward-time=0, network-operator-premium-activation=1, video-device-blacklisted=0, collection=1, fb-grant-permission-local-render=0, backend-advised-bitrate=1, catalogue=premium, storage-size-config=10240,90,500,3, ads=0, head-files-url=https://heads-fa.scdn.co/head/{file_id}, ab-collection-followed-artists-only=0, browse-overview-enabled=1, on-demand=1, ap-resolve-pods=0, name=Spotify Premium, sidebar-navigation-enabled=0, loudness-levels=1:-5.0,0.0,3.0:-2.0, mobile-login=1, hifi-optin-intent=0, preferred-locale=en, license-acceptance-grace-days=30, prefetch-strategy=18, ugc-abuse-report=1, ab-watch-now=0, financial-product=pr:premium,tc:0,rt:v2_NL_default_new-family-master_14.99_EUR_default, ab-collection-offline-mode=0, streaming-rules=, head-files=1, capping-bar-threshold=3601, ab-new-share-flow=0, video-keyframe-url=http://keyframes-fa.cdn.spotify.com/keyframes/v1/sources/{source_id}/keyframe/heights/{height}/timestamps/{timestamp_ms}.jpg, mobile=1, is_maybe_in_social_session=0, ab-mobile-startpage=0, ab-moments-experience=0, ab-mobile-social-feed=1, ab-test-group=593, player-license=premium}
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,775 INFO  Session:1415 - Skipping UnknownData_AllZeros
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,776 INFO  Session:1415 - Skipping LegacyWelcome
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,779 TRACE MercuryClient:199 - Handling packet, cmd: MercuryEvent, seq: 5663893357441581056, flags: 1, parts: 1
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,784 DEBUG MercuryClient:234 - Couldn't dispatch Mercury event {seq: 5663893357441581056, uri: hm://pusher/v1/connections/NDk3YzE5N2UxMWU0NTZmYTZlY <<SNIP>> UQ1ODdBRDkyRQ%3D%3D, code: 200, payload: []}
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,786 DEBUG Session:1391 - Received 0x10: DA56C9795514FB24BAD01FEEA25662A4E8E1F5D1
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,850 TRACE MercuryClient:199 - Handling packet, cmd: MercuryReq, seq: 1, flags: 1, parts: 2
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,854 DEBUG TokenProvider:71 - Updated token successfully! {scopes: [playlist-read], newToken: StoredToken{expiresIn=3600, accessToken='BQCkMm...4WoiyY', scopes=[playlist-read], timestamp=1683990028850}}
May 13 17:00:28 livingroom java[3785]: 2023-05-13 17:00:28,892 INFO  Session:361 - Authenticated as mxyes06jwepj16ar75nv2msh3!
May 13 17:00:29 livingroom java[3785]: 2023-05-13 17:00:29,090 INFO  Session:1342 - Skipping unknown command {cmd: 0x75, payload: 000000}
May 13 17:00:29 livingroom java[3785]: 2023-05-13 17:00:29,180 TRACE AsyncProcessor:52 - AsyncProcessor{put-state-worker} has started
May 13 17:00:29 livingroom java[3785]: 2023-05-13 17:00:29,633 DEBUG DealerClient:348 - Dealer connected! {host: guc3-dealer.spotify.com}
May 13 17:00:29 livingroom java[3785]: 2023-05-13 17:00:29,942 DEBUG DeviceStateHandler:155 - Updated Spotify-Connection-Id: NDk3YzE <<SNIP>> NjQ3RUFEQkE0
May 13 17:00:29 livingroom java[3785]: 2023-05-13 17:00:29,974 INFO  StateWrapper:344 - Notified new device (us)!
May 13 17:00:30 livingroom java[3785]: 2023-05-13 17:00:30,613 INFO  DeviceStateHandler:273 - Put state. {ts: 1683990029963, connId: NDk3Y...EQkE0, reason: NEW_DEVICE, request: device { device_info { can_play: true volume: 65536 name: "livingroomwq" capabilities { can_be_player: true gaia_eq_connect_id: true supports_logout: true is_observable: true volume_steps: 64 supported_types: "audio/episode" supported_types: "audio/track" command_acks: true supports_playlist_v2: true is_controllable: true supports_transfer_command: true supports_command_request: true supports_gzip_pushes: true } device_software_version: "librespot-java 1.6.2" device_type: COMPUTER spirc_version: "3.2.6" device_id: "497c197e11e456fa6ecca32333ed4fc720925176" client_id: "65b708073fc0480ea92a077233ca87bd" } player_state { context_restrictions { } playback_speed: 1.0 position_as_of_timestamp: 0 is_system_initiated: true options { shuffling_context: false repeating_context: false repeating_track: false } suppressions { } } } member_type: CONNECT_STATE put_state_reason: NEW_DEVICE has_been_playing_for_ms: 0 client_side_timestamp: 1683990029963}
May 13 17:00:30 livingroom java[3785]: 2023-05-13 17:00:30,775 TRACE DealerClient:114 - Received request. {mid: hm://connect-state/v1/player/command, key: 64846ad7-68a1-4174-a78b-aef457eec337, pid: 362851302, sender: bea92c5c6d1f44e4d7736b645ef06e0d61a32957, command: {"endpoint":"transfer","data":"CgYIABAAGAASlQYIytn6rIExEKiyBhkA <<SNIP>>ADIAIgIQAA==","options":{"restore_paused":"restore","restore_position":"extrapolate","restore_track":"always_play_something","license":"premium"},"from_device_identifier":"bea92c5c6d1f44e4d7736b645ef06e0d61a32957"}}
May 13 17:00:30 livingroom java[3785]: 2023-05-13 17:00:30,786 DEBUG Player:106 - Received command: Transfer
May 13 17:00:30 livingroom java[3785]: #
May 13 17:00:30 livingroom java[3785]: # A fatal error has been detected by the Java Runtime Environment:
May 13 17:00:30 livingroom java[3785]: #
May 13 17:00:30 livingroom java[3785]: #  SIGBUS (0x7) at pc=0xf766f730, pid=3785, tid=3834
May 13 17:00:30 livingroom java[3785]: #
May 13 17:00:30 livingroom java[3785]: # JRE version: OpenJDK Runtime Environment (11.0.18+10) (build 11.0.18+10-post-Raspbian-1deb11u1)
May 13 17:00:30 livingroom java[3785]: # Java VM: OpenJDK Server VM (11.0.18+10-post-Raspbian-1deb11u1, mixed mode, g1 gc, linux-)
May 13 17:00:30 livingroom java[3785]: # Problematic frame:
May 13 17:00:30 livingroom java[3785]: # V  [libjvm.so+0x9b9730]
May 13 17:00:30 livingroom java[3785]: #
May 13 17:00:30 livingroom java[3785]: # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
May 13 17:00:30 livingroom java[3785]: #
May 13 17:00:30 livingroom java[3785]: # An error report file with more information is saved as:
May 13 17:00:30 livingroom java[3785]: # /opt/spocon/hs_err_pid3785.log
May 13 17:00:30 livingroom java[3785]: #
May 13 17:00:30 livingroom java[3785]: # If you would like to submit a bug report, please visit:
May 13 17:00:30 livingroom java[3785]: #   Unknown
May 13 17:00:30 livingroom java[3785]: #
May 13 17:00:30 livingroom systemd[1]: spocon.service: Main process exited, code=killed, status=6/ABRT
May 13 17:00:30 livingroom systemd[1]: spocon.service: Failed with result 'signal'.
May 13 17:00:30 livingroom systemd[1]: spocon.service: Consumed 12.293s CPU time.
May 13 17:00:40 livingroom systemd[1]: spocon.service: Scheduled restart job, restart counter is at 2.
May 13 17:00:40 livingroom systemd[1]: Stopped SpoCon.
May 13 17:00:40 livingroom systemd[1]: spocon.service: Consumed 12.293s CPU time.
May 13 17:00:40 livingroom systemd[1]: Starting SpoCon...
```

**Additional context**
I am running the same version of spocon on another raspberry pi under linux 5 and it appears to work properly.

**Comment by txoof** [comment]
Here's the log output from `/opt/spocon/`
[hs_err_pid867.log](https://github.com/spocon/spocon/files/11470452/hs_err_pid867.log)

**Comment by spocon** [comment]
new version should fix this issue: 1.6.3 released
